vyos/vyos-1x
# openvpn: T56: fix username/password authentication

## Change Summary


Fixed bug where 'tls auth_file' was checked but this should be 'authentication' for username/password authentication.

If someone configured an openvpn client with username/password only it would not work as it will still check if 'tls cert-file' and 'tls key-file' are specified.

This change matches the older Perl script here: https://github.com/vyos/vyatta-openvpn/blob/current/lib/Vyatta/OpenVPN/Config.pm#L768

## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T56

## Component(s) name


openvpn

## Proposed changes


Fix the check so openvpn username/password authentication only works again

## How to test


This was not allowed (but should):
```
set interfaces openvpn vtun1 mode client
set interfaces openvpn vtun1 remote-host 127.0.0.1
set interfaces openvpn vtun1 remote-port 1234
set interfaces openvpn vtun1 tls ca-cert-file /config/auth/vpn/ca.crt
set interfaces openvpn vtun1 authentication password 'y'
set interfaces openvpn vtun1 authentication username 'x'
```

After this patch it works.

## Checklist:
